src/include/pka_cpu.h

```diff
@@ -79,17 +85,212 @@ static inline uint64_t pka_early_cpu_speed(void)
   return CPU_HZ_MAX;
 }
 #else
-
+#define PKA_SYS_SMBIOS_ENTRY_POINT "/sys/firmware/dmi/tables/smbios_entry_point"
+#define PKA_SYS_DMI_TABLE "/sys/firmware/dmi/tables/DMI"
+#define PKA_DMI_HEADER_TYPE_OFFSET 0
+#define PKA_DMI_HEADER_LEN_OFFSET 1
+#define PKA_DMI_HEADER_HANDLE_OFFSET 2
+#define PKA_SMBIOS_LEN_OFFSET 0x06
+#define PKA_SMBIOS_EXPECTED_LEN 0x20
+#define PKA_DMI_TABLE_OFFSET 0x0C
+#define PKA_DMI_PROC_FREQ_OFFSET 0x16
+#define PKA_DMI_PROC_TABLE_LEN 0x1A
+#define PKA_DMI_PROC_TYPE 4
+#define PKA_DMI_TABLE_LEN 4
+#define PKA_DMI_TYPE_END 127
+#ifdef __BIG_ENDIAN__
+#define WORD(x) (u16)((x)[0] + ((x)[1] << 8))
+#define DWORD(x) (u32)((x)[0] + ((x)[1] << 8) + ((x)[2] << 16) + ((x)[3] << 24))
+#define QWORD(x) (U64(DWORD(x), DWORD(x + 4)))
+#else
+#define WORD(x) (uint16_t)(*(const uint16_t *)(x))
+#define DWORD(x) (uint32_t)(*(const uint32_t *)(x))
+#define QWORD(x) (*(const uint64_t *)(x))
+#endif
 #define MAX_CLOCK_CYCLES     UINT64_MAX
 
+typedef struct {
+    uint16_t handle;
+    uint8_t  type;
+    uint8_t  length;
+    uint8_t *data;
+} pka_cpu_dmi_hdr_t;
+
+/// Global variable holding the cpu frequency.
+uint64_t cpu_f_hz;
+
+static void *pka_cpu_read_sysfs(size_t *file_len, const char *filename)
+{
+    struct stat sbuf;
+    size_t      read_count;
+    ssize_t     read_status;
+    uint8_t    *file_mem;
+    int         fd;
+
+    read_count  = 0;
+    read_status = 1;
+
+    fd = open(filename, O_RDONLY);
+    if (fd == -EPERM) {

```

Comment:
Fyi, to be consistent with the PKA lib coding style '{' and must be on the next line:

  if (condition)
  {
     statement1;
     statement2;
  }

and

  if (condition)
  {
     statement1;
  }
  else
  {
     statement2;
     statement3;
  }